File: app/controllers/utils/AuthController.java
package controllers.utils;

//import akka.japi.Option;
import play.Application;
import play.api.libs.json.*;
import play.mvc.*;
import scala.Option;
import scala.Some;
import securesocial.core.*;
import service.UserService;
import securesocial.core.AuthenticationMethod;
import securesocial.core.Identity;
import securesocial.core.IdentityId;
import securesocial.core.OAuth2Info;
import securesocial.core.PasswordInfo;
import securesocial.core.SocialUser;
import securesocial.core.java.BaseUserService;
import securesocial.core.java.Token;

public class AuthController extends Controller {
	
/*	public Result authenticateMobile(String providerName) {
		
		OAuth2Info oauth2Info = new OAuth2Info( request().body().asJson().toString(), null, null, null);
		IdentityProvider provider = Registry.providers().get(providerName).get();
		
		SocialUser filledUser = provider.fillProfile(
				new SocialUser(new IdentityId("", provider.id()), 
						"", 
						"", 
						"", 
						Option.apply(""), 
						Option.apply(""), 
						provider.authMethod(), 
						null, 
						Option.apply(oauth2Info), 
						null) 
				);
				
		UserService.doFind(filledUser.identityId());
			
		
		
	}*/
	
}

File: app/controllers/auth/AuthController.scala
package controllers.auth

import play.api.libs.json.Json
import play.api.mvc._
import securesocial.core._

class AuthController extends Controller {
  private implicit val readsOAuth2Info = Json.reads[OAuth2Info]
  // Some of the below code is taken from ProviderController in SecureSocial
  def authenticateMobile(providerName: String) = Action(parse.json) { implicit request =>
    // format: { "accessToken": "..." }
    val oauth2Info = request.body.asOpt[OAuth2Info]
    val provider = Registry.providers.get(providerName).get
    val filledUser = provider.fillProfile(
      SocialUser(IdentityId("", provider.id), "", "", "", None, None, provider.authMethod, oAuth2Info = oauth2Info))
    UserService.find(filledUser.identityId) map { user =>
      val newSession = Events.fire(new LoginEvent(user)).getOrElse(session)
      Authenticator.create(user).fold(
        error => throw error,
        authenticator => Ok(Json.obj("sessionId" -> authenticator.id))
          .withSession(newSession - SecureSocial.OriginalUrlKey - IdentityProvider.SessionId - OAuth1Provider.CacheKey)
          .withCookies(authenticator.toCookie)
      )
    } getOrElse NotFound(Json.obj("error" -> "user not found"))
  }
  // any other methods you might have relating to authentication ...
}
